modules/localization/msf/common/util/base_map_cache_test.cc

```diff
@@ -0,0 +1,154 @@
+/******************************************************************************
+ * Copyright 2020 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+#include "modules/localization/msf/common/util/base_map_cache.h"
+#include <vector>
+#include "gtest/gtest.h"
+
+namespace apollo {
+namespace localization {
+namespace msf {
+
+class NodeIndex {
+ public:
+  NodeIndex() : m_(0), n_(0) {}
+  NodeIndex(int m, int n) : m_(m), n_(n) {}
+
+ public:
+  bool operator<(const NodeIndex& index) const {
+    return std::forward_as_tuple(m_, n_) <
+           std::forward_as_tuple(index.m_, index.n_);
+  }
+  bool operator==(const NodeIndex& index) const {
+    return m_ == index.m_ && n_ == index.n_;
+  }
+  bool operator!=(const NodeIndex& index) const {
+    return m_ != index.m_ || n_ != index.n_;
+  }
+
+ private:
+  int m_;
+  int n_;
+};
+
+class NodeData {
+ public:
+  NodeData() : is_reserved_(false), name_("") {}
+  explicit NodeData(std::string name) : is_reserved_(false), name_(name) {}
+
+ public:
+  void SetIsReserved(bool b) { is_reserved_ = b; }
+  bool GetIsReserved() { return is_reserved_; }
+
+  void SetName(std::string name) { name_ = name; }
```

Comment:
`const std::string& name` ?

Reply:
> `const std::string& name` ?

OK

Reply:
Actually "string_view" is the new recommendation. But it's OK to stick to `const string&` for a while.